src/NHibernate.Test/Linq/ConstantTest.cs

```diff
@@ -257,6 +259,41 @@ public void PlansAreCached()
 			}
 		}
 
+		//GH-2298 - Different Update queries - same query cache plan
+		[Test]
+		public void DmlPlansAreProperlyHandled()
```

Comment:
This test would be better put into `LinqBulkManipulation` namespace I think.

Reply:
I prefer to keep it here. IMHO it's very similar to `PlansWithNonParameterizedConstantsAreNotCachedForExpandedQuery` only for DML query. Also I think it's better to keep tests with a bit hacky reflection logic in one place (part about retrieving query plan cache).